gangplank/ocp/cosa-podman.go

```diff
@@ -248,3 +251,37 @@ func podmanRunner(term termChan, cp CosaPodder, envVars []v1.EnvVar) error {
 	}
 	return nil
 }
+
+// mustHaveImage pulls the image if it is not found
+func mustHaveImage(ctx context.Context, image string) error {
+	uri, _ := os.LookupEnv(podmanContainerHostEnvVar)
+	auth, authDefined := os.LookupEnv(podmanContainerHostSSHEnvVar)
+	if !authDefined {
+		auth, authDefined = os.LookupEnv("SSH_AUTH_SOCKET")
+	}
+
+	// remote podman requires either the CONTAINER_SSHKEY or SSH_AUTH_SOCKET be defined.
+	isRemote := strings.HasPrefix(uri, "ssh://")
+	if isRemote && !authDefined {
+		return fmt.Errorf("remote podman requires %s to be defined", podmanContainerHostSSHEnvVar)
+	}
+
+	if isRemote {
+		var err error
+		ctx, err = bindings.NewConnectionWithIdentity(ctx, uri, auth)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
```suggestion
```

We were provided a `ctx` which is a podman connection that already
exists (created in `podmanRunner` and passed to `mustHaveImage`).
There's no reason to create a new connection.